backend/src/repository/history-transformer.repository.ts

```diff
@@ -0,0 +1,72 @@
+import { Database } from 'better-sqlite3';
+import { TransformerDTO, TransformerFilterDTO } from '../../../shared/model/transformer.model';
+export const HISTORY_TRANSFORMERS_TABLE = 'history_transformers';
+
+export default class HistoryTransformerRepository {
+  constructor(private readonly database: Database) {}
+
+  /**
+   * Adds a transformer to the list of available transformers for a history connector south or north
+   */
+  addTransformer(historyId: string, connectorType: 'south' | 'north', transformerId: string): void {
+    if (connectorType !== 'south' && connectorType !== 'north') {
+      throw new Error('Invalid connectory type');
+    }
+
+    this.database
+      .prepare(`INSERT INTO ${HISTORY_TRANSFORMERS_TABLE} (history_id, connector_type, transformer_id) VALUES (?, ?, ?)`)
+      .run(historyId, connectorType, transformerId);
+  }
+
+  /**
+   * Retrieves all transformers available for a history connector south or north
+   * @param filter Optional filter to restrict the results
+   */
+  getTransformers(historyId: string, connectorType: 'south' | 'north', filter?: TransformerFilterDTO): Array<TransformerDTO> {
+    if (connectorType !== 'south' && connectorType !== 'north') {
+      throw new Error('Invalid connectory type');
+    }
+
+    if (!filter) {
+      const query = `SELECT t.id, t.name, t.description, t.input_type AS inputType, t.output_type AS outputType, t.code, t.file_regex AS fileRegex FROM ${HISTORY_TRANSFORMERS_TABLE} ht
+        JOIN transformers t ON ht.transformer_id = t.id
+        WHERE ht.history_id = ? AND ht.connector_type = ?;`;
+      return this.database.prepare(query).all(historyId, connectorType) as Array<TransformerDTO>;
+    }
+
+    let query = `SELECT t.id, t.name, t.description, t.input_type AS inputType, t.output_type AS outputType, t.code, t.file_regex AS fileRegex FROM ${HISTORY_TRANSFORMERS_TABLE} ht
+      JOIN transformers t ON ht.transformer_id = t.id`;
```

Comment:
Shouldn't we use the TRANSFORMERS_TABLE constant instead?

Reply:
Yes, will do